src/pedlar.ts

```diff
@@ -66,15 +122,55 @@ export class Pedlar {
    * @param handler The handler that should be invoked when the event is emitted
    * @param options Event listener options
    */
-  public addEvent<K extends keyof HTMLElementEventMap>(
-    el: HTMLElement,
-    eventType: K,
+  public addEvent(
+    el: Element,
+    eventType: keyof HTMLElementEventMap,
     handler: EventListenerOrEventListenerObject,
     options?: boolean | EventListenerOptions,
-  ): string {
+  ): PedlarEffect {
     return this.perform(() => {
       el.addEventListener(eventType, handler, options)
       return () => el.removeEventListener(eventType, handler, options)
     })
   }
 }
+
+function validateDestroyer(destroyer: Function) {
+  if (typeof destroyer === 'function') return
+  throw new Error(
+    'Effect must return undefined, null, or a function; found: ' +
+      typeof destroyer,
+  )
+}
+
+/**
+ * @returns `true` if one of the values in the two passed arrays
```

Comment:
This line doesn’t really make sense. True if the values at all equivalent indexes in both arrays are found to be equal